From 9e0b53f9629734b6ca516b805cbd1ddfc6ac3ec9 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Fri, 14 Apr 2023 00:24:47 -0700
Subject: [PATCH 58/60] UBUNTU: SAUCE: apparmor3.2.0 [58/60]: fix: add missing
 failure check in compute_xmatch_perms

This fixes a patch in the out of tree apparmor prompting patchset that
ubuntu carries.

Add check for failure to allocate the permission table.

BugLink: http://bugs.launchpad.net/bugs/2017903
Fixes: caa9f579ca72 ("apparmor: isolate policy backwards compatibility to its own file")
Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 security/apparmor/policy_compat.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/security/apparmor/policy_compat.c b/security/apparmor/policy_compat.c
index d5ee9f6e5763..0cb02da8a319 100644
--- a/security/apparmor/policy_compat.c
+++ b/security/apparmor/policy_compat.c
@@ -182,6 +182,8 @@ static struct aa_perms *compute_xmatch_perms(struct aa_dfa *xmatch,
 	state_count = xmatch->tables[YYTD_ID_BASE]->td_lolen;
 	/* DFAs are restricted from having a state_count of less than 2 */
 	perms = kvcalloc(state_count, sizeof(struct aa_perms), GFP_KERNEL);
+	if (!perms)
+		return NULL;
 	*size = state_count;
 
 	/* zero init so skip the trap state (state == 0) */
-- 
2.40.1

